vulcan/lib/server/workflows/scripts/umap_finalize_downloadable_object.py

```diff
@@ -1,17 +1,20 @@
-from archimedes.functions.dataflow import input_path, output_json, input_json, output_path, buildTargetPath
+from archimedes.functions.dataflow import input_path, output_json, input_json, output_path, buildTargetPath, parseModelAttr
 from archimedes.functions.magma import connect, question
 from archimedes.functions.scanpy import scanpy as sc
 from archimedes.functions.environment import project_name
 from archimedes.functions.utils import re
+from archimedes.functions.list import unique
 
 scdata = sc.read(input_path('scdata.h5ad'))
+pdat = input_json("project_data")[project_name]
 
 # Prep magma querying
+seq_model = parseModelAttr(pdat['seq_h5_counts_data'])['model']
 magma = connect()
 def get(ids, value):
     ids = ids if type(ids) == list else ids.tolist()
     values = dict(question(magma, [
-        'sc_seq',
+        seq_model,
```

Comment:
> Looks pretty neat! I noticed that sometimes (for IPI), the Record Selection step shows `null` for Tissue and Compartment. Is that supposed to be a valid thing that a user might want to select on? I assume it shows up because the field isn't filled in on the Magma side, but wasn't sure if users would expect to be able to perform analysis on samples where those values are `null`.
> 
> Also, I'm getting an error in the Prep output anndata object ... not sure if it's because I've chosen a set of invalid / conflicting options, or a bug?
> 
> ```
> Python error while executing script: Traceback (most recent call last):
>   File "/archimedes-exec/tmpc42s3fl6/script.py", line 27, in <module>
>     data = get(scdata.obs[ 'Record_ID' ].values, buildTargetPath(color_options[color_by], pdat))
>   File "/archimedes-exec/tmpc42s3fl6/script.py", line 13, in get
>     values = dict(question(magma, [
>   File "/app/archimedes/functions/magma/magma.py", line 14, in question
>     raise Exception('No answer to magma query')
> Exception: No answer to magma query
> ```
> 
> I can send you my JSON settings file or attach to this PR, if that helps clarify what the issue is.
> 
> Note that I've tried this with various combinations of Kidney / Hepatobiliary samples, and cannot get past the `Prep output anndata object` step... cry

Mannnnn, that took me a while, and the real issue was just a hard coded reference to an 'sc_seq' model.  Not sure how this hard-code came back as we both tested with ipi before!


Anyway, the code down below here makes the querying of magma a bit less demanding anyway, and if an issue arises from this query in future, I've added some context export to the exception that gets raised.

Reply:
Oh, tricky find! Let me test!